Add HTTP Referer to requests for Technicolor Routers

macearl opened the pull request:
Adds HTTP Referer Header for all Technicolor Requests to make it compatible with Technicolor CGA4322DE Firmware: 2.0.17-IMS-KDG

Without it the API returns an error:
```
Slim Application Error

The application could not run because of the following error:
Details
Type: ErrorException
Code: 8
Message: Undefined index: HTTP_REFERER
File: /usr/www/api/v1/session.php
Line: 69
Trace

#0 /usr/www/api/v1/session.php(69): Slim\Slim::handleErrors(8, 'Undefined index...', '/usr/www/api/v1...', 69, Array)
#1 [internal function]: {closure}()
#2 /usr/www/api/Slim/Route.php(468): call_user_func_array(Object(Closure), Array)
#3 /usr/www/api/Slim/Slim.php(1357): Slim\Route->dispatch()
#4 /usr/www/api/Slim/Middleware/Flash.php(85): Slim\Slim->call()
#5 /usr/www/api/Slim/Middleware/MethodOverride.php(92): Slim\Middleware\Flash->call()
#6 /usr/www/api/Slim/Middleware/PrettyExceptions.php(67): Slim\Middleware\MethodOverride->call()
#7 /usr/www/api/Slim/Slim.php(1302): Slim\Middleware\PrettyExceptions->call()
#8 /usr/www/api/index.php(136): Slim\Slim->run()
#9 {main}
```

also increases the version number and includes the new firmware version to the README

macearl commented:
not sure why the CI failed, i'm not too familiar with the setup on github, are there some logs available?

an0nfunc commented:
You can find them here: https://github.com/totev/vodafone-station-cli/runs/4720391108?check_suite_focus=true#step:5:9

totev commented on src/modem/technicolor-modem.ts:
```diff
@@ -151,6 +151,7 @@ export class Technicolor extends Modem {
       const {data: salt} = await this.httpClient.post<TechnicolorSaltResponse>('/api/v1/session/login', `username=${Modem.USERNAME}&password=seeksalthash`, {
         headers: {
           'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'},
+          'Referer': 'http://${this.modemIp}/',
```
This should be inside the **headers }** :)

totev reviewed (changes requested):
Thanks for the update! Once you've moved up the one Referer into the headers, I'll happily merge it and release a new version.

totev commented:
@an0nfunc thanks for the link!
@macearl I wrote an inline comment, I think that's why the CI is complaining :)

macearl commented:
> You can find them here: https://github.com/totev/vodafone-station-cli/runs/4720391108?check_suite_focus=true#step:5:9

thanks, i actually found that link yesterday, however i could not see the logs due to my script blocker ;)



> Thanks for the update! Once you've moved up the one Referer into the headers, I'll happily merge it and release a new version.

whoops, fixed the curly brace position. I also got another update tonight so i also added `3.0.41-IMS-KDG` as compatible
